src/mds/Server.cc

```diff
@@ -3372,18 +3372,20 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
   _inode->truncate_seq = 1; /* starting with 1, 0 is kept for no-truncation logic */
 
   CInode *diri = dir->get_inode();
+  const CInode::mempool_inode *pip = diri->get_projected_inode().get();
```

Comment:
nit: use `auto`.

Reply:
also, any reason not to work with just a `CInode *`?

Reply:
> nit: use `auto`.

Sure, will fix it.

> also, any reason not to work with just a `CInode *`?

Sorry, didn't get it here. What do you mean `CInode *` ? Do you mean `diri->get_inode()` ?